### PR TITLE
Add documentation for check-capacity Provisioning class

### PR DESCRIPTION
**Pull request by yaroslava-serdiuk**
#### What type of PR is this?

/kind documentation

#6814

### CODE REVIEW

File: cluster-autoscaler/FAQ.md
Comment (alculquicondor) [inline cluster-autoscaler/FAQ.md:571]:
```suggestion
`--enable-provisioning-request=true`.
```

**Review by alculquicondor — COMMENTED** [review]
/lgtm

File: cluster-autoscaler/FAQ.md
Comment (kisieland) [inline cluster-autoscaler/FAQ.md:564]: Please split the line into more, this should not influence how the result is rendered.

**Review by kisieland — APPROVED** [review]
One nit otherwise
/lgtm

File: cluster-autoscaler/FAQ.md
Comment (cortespao) [inline cluster-autoscaler/FAQ.md:151]: The term "decent" might be subjective. I recommend that we use more factual information, if possible.

File: cluster-autoscaler/FAQ.md
Comment (cortespao) [inline cluster-autoscaler/FAQ.md:566]:
```suggestion
ProvisioningRequest (or ProvReq) is a new namespaced Custom Resource that allows you to ask CA for capacity for groups of pods. For a detailed explanation of the ProvisioningRequest API, please refer to the
[proposal](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/proposals/provisioning-request.md).

```

File: cluster-autoscaler/FAQ.md
Comment (cortespao) [inline cluster-autoscaler/FAQ.md:572]:
```suggestion
`--enable-provisioning-reques=true`.
```

File: cluster-autoscaler/FAQ.md
Comment (cortespao) [inline cluster-autoscaler/FAQ.md:576]:
```suggestion
ClusterAutoscaler supports the following ProvisioningClasses:
```

File: cluster-autoscaler/FAQ.md
Comment (cortespao) [inline cluster-autoscaler/FAQ.md:588]:
```suggestion
  Adds a `Accepted=True` condition when ProvReq is accepted by ClusterAutoscaler and ClusterAutoscaler will check capacity for this ProvReq.
  Adds a `Provisioned=True` condition to the ProvReq if capacity is available.
  Adds a `BookingExpired=True` condition when the 10-minute reservation period expires.
```

**Review by cortespao — COMMENTED** [review]
Sending a couple of suggestions. Thanks!

File: cluster-autoscaler/FAQ.md
Comment (yaroslava-serdiuk) [reply cluster-autoscaler/FAQ.md:151]: This is an automated formatting change and doesn't belong to the actual documentation that I added

File: cluster-autoscaler/FAQ.md
Comment (MaciekPytel) [inline cluster-autoscaler/FAQ.md:565]: Is that up to date? IIRC we've made some changes from the original design during the implementation?

File: cluster-autoscaler/FAQ.md
Comment (MaciekPytel) [inline cluster-autoscaler/FAQ.md:583]: Should we stress it out more that this is just 'reserving' in the sense of changing outcome of other check-capacity requests? It's not reserving it from other pods that may be created.
You say that at the end of the sentence, but I think it's worth highlighting. To me a default interpretation of reserving capacity would be a guarantee that no-one else will use it - but that's only true if everyone is blocking their job creation on check-capacity requests (i.e. in practice this only works for Kueue and no-one else).

File: cluster-autoscaler/FAQ.md
Comment (MaciekPytel) [inline cluster-autoscaler/FAQ.md:1063]: nit: Should this be formatted that way? This is a copy-paste from kubectl output (ie. preformatted text), not markdown

**Comment by MaciekPytel** [comment]
/approve
/hold
Left a few comments, adding hold to allow addressing them. Please feel free to remove hold without additional review from me if I'm not responsive enough

File: cluster-autoscaler/FAQ.md
Comment (yaroslava-serdiuk) [reply cluster-autoscaler/FAQ.md]: Right, updated to the previous version.

File: cluster-autoscaler/FAQ.md
Comment (yaroslava-serdiuk) [reply cluster-autoscaler/FAQ.md]: Thanks, added stress that we reserve capacity from other ProvReqs

File: cluster-autoscaler/FAQ.md
Comment (yaroslava-serdiuk) [reply cluster-autoscaler/FAQ.md]: I think implementation details are not correct, but the API didn't change except of suffix, so I think it's ok to refer this proposal for detailed explanation of the ProvisioningRequest API. 
I added "original proposal" to emphasise that some other details of the proposal could be outdated.

File: cluster-autoscaler/FAQ.md
Comment (alculquicondor) [reply cluster-autoscaler/FAQ.md]: Just for my understanding (it doesn't have to go in the documentation):
It also blocks individual Pods that cluster-autoscaler processes from using that capacity. But because those pods would likely just be scheduled by kube-scheduler, in practice nothing prevents them from taking it.
Is that understanding correct?

Comment (MaciekPytel) [reply cluster-autoscaler/FAQ.md]: That's not true, at list not how it's implemented in #6880 right now - you're only booking capacity _after_ CA assigns all the pending pods to the nodes (which happens in FilterOutSchedulable / PodListProcessor) and you're only booking it until the end of CA loop. Meaning that any pending pod that shows up next loop will be allocated to this capacity before you re-book it.
As-is the implementation is only really preventing scale-down. To reserve from pods as you described the booking logic would need to happen before FilterOutSchedulable, not after it.

Comment (yaroslava-serdiuk) [reply cluster-autoscaler/FAQ.md]: You're right, thanks for correcting me! 
Aldo, your understanding is correct, it doesn't make a lot of sence to block CA from using the capacity, since CA just tries to predict behaviour of scheduler.

**Pull request merged**
